packages/zk-transfer-web/src/components/Token.tsx

```diff
@@ -1,31 +1,55 @@
 // @ts-ignore
-import React, { useState, useEffect, useRef } from "react";
+import React, { useState, useCallback, useEffect, useRef } from "react";
 import arrowLeft from "../assets/arrow-left.svg";
+// @ts-ignore
+import { note } from "aztec.js"; 
+import { NoteList, NoteValue } from "../libs/types";
+import { getSum } from "../util";
 
 type TokenPropType = {
-  firstValue: number;
-  secondValue: number;
+  name: string;
+  firstValue?: number;
+  secondValue?: number;
   children: any;
   canEdit?: boolean;
+  notes?: NoteList;
   onSend: Function;
   allowEdit?: Function;
 };
 
+const fetchValue = (notes: NoteList) => {
+  const getNoteValue = function (e: any) {
+    return note.fromViewKey(e.get("viewingKey"))
+  };
+  const pendingValues: NoteValue[] = notes.map(getNoteValue);
+  return Promise.all(Array.from(pendingValues));
+}
+
 const TokenInput = (props: TokenPropType) => {
-  const editMode = props.canEdit;
-  const [digit, setDigit] = useState(props.firstValue);
   const input = useRef(null);
+  const editMode = props.canEdit;
+  const [digit, setDigit] = useState(0);
+  const [fetched, setFetched] = useState(false);
 
-  const setEditMode = (a?: boolean) => props.allowEdit && props.allowEdit();
-  const canEdit = (cb: Function) => (...args: any[]) =>
-    props.canEdit && cb(...args);
+  const setEditMode = useCallback((newEditStatus?: boolean) => {
+    props.allowEdit && props.allowEdit(newEditStatus)
+  }, [props]);
+  const canEdit = useCallback((cb: Function) => (...args: any[]) =>
+    props.canEdit && cb(...args), [props.canEdit]);
 
   useEffect(() => {
     canEdit((a: any) => {
-      // console.log('The input at first', a);
       if (a) a.current.focus();
     })(input);
-  });
+    if (!fetched && props.notes) {
+       console.log('Fetching balances for ', props.name);
+       fetchValue(props.notes)
+        .then(values => setDigit(values.reduce(getSum, 0)));
```

Comment:
It's okay. Because i've used it only one time.

Reply:
ok, please make the change, thanks.

Reply:
When i try this `parseInt(v, 16)`, i get `33` instead of `22` @cryptogoth

Reply:
You mean 34? That's the decimal value for the hex value 22.
However, you're parsing correctly in your code, since minting of a 10-value note shows up correct as a (decimal) 10 balance, not as 16 (which would be the value if we converted it as `parseInt('10', 16)`